mono/utils/mono-proclib.c

```diff
@@ -879,6 +881,23 @@ mono_cpu_count (void)
 	/* FIXME: warn */
 	return 1;
 }
+
+/**
+ * mono_cpu_limit:
+ * \returns the number of processors available to this process
+ */
+int
+mono_cpu_limit (void)
+{
+	int count = 0,
+	    max;
```

Comment:
Can you add support here also for setting the count using an envvar (https://github.com/dotnet/runtime/issues/60915)?

When `DOTNET_PROCESSOR_COUNT` envvar is set to a value `>0` it should be returned.
It must not be truncated to `mono_cpu_count`.

Can/should we cache this value, like coreclr?

Reply:
To confirm from other issues in this repo, this value is decimal.